htmd/parameterization/fftype.py

```diff
@@ -125,112 +140,85 @@ def fftype(mol, rtfFile=None, prmFile=None, method='GAFF2', acCharges=None, tmpD
-                                                              ','.join(np.setdiff1d(acmol.name, mol.name))
-                                                              )
-                                       )
-
                 # Read the results
                 prm = parmed.amber.AmberParameterSet(os.path.join(tmpdir, 'mol.frcmod'))
-                names, atomtypes, charges, impropers = readPREPI(mol, os.path.join(tmpdir, 'mol.prepi'))
+                names, atomtypes, charges, impropers = readPREPI(renamed_mol, os.path.join(tmpdir, 'mol.prepi'))
                 masses, elements = readFRCMOD(atomtypes, os.path.join(tmpdir, 'mol.frcmod'))
+
             elif method == 'CGenFF_2b6':
 
                 # Write the molecule to a file
-                mol.write(os.path.join(tmpdir, 'mol.pdb'))
+                renamed_mol.write(os.path.join(tmpdir, 'mol.pdb'))
 
                 # Run match-type
-                returncode = subprocess.call([match_binary,
-                                              '-charge', str(netcharge),
-                                              '-forcefield', 'top_all36_cgenff_new',
-                                              'mol.pdb'], cwd=tmpdir)
+                cmd = ['match-typer',
+                       '-charge', str(netcharge),
+                       '-forcefield', 'top_all36_cgenff_new',
+                       'mol.pdb']
+                returncode = subprocess.call(cmd, cwd=tmpdir)
                 if returncode != 0:
                     raise RuntimeError('"match-typer" failed')
 
                 prm = parmed.charmm.CharmmParameterSet(os.path.join(tmpdir, 'mol.rtf'), os.path.join(tmpdir, 'mol.prm'))
                 names, elements, atomtypes, charges, masses, impropers = readRTF(os.path.join(tmpdir, 'mol.rtf'))
-                # addParmedResidue(prm, names, elements, atomtypes, charges, impropers)
+
             else:
                 raise ValueError('Invalid method {}'.format(method))
 
-    # Substituting values from the read-in topology
+        assert np.all(renamed_mol.name == names)
+
+    assert np.all(mol.element == elements)
+
     mol = mol.copy()
-    mol.name = names
-    mol.element = elements
     mol.atomtype = atomtypes
+    mol.masses = masses
+    mol.impropers = impropers
     if acCharges is not None:
         mol.charge = charges
-    mol.impropers = impropers
-    if np.sum(mol.masses) == 0:
-        mol.masses = masses
 
     return prm, mol
 
 
 class TestFftype(unittest.TestCase):
 
-    def __init__(self, *args, **kwargs):
-        super(TestFftype, self).__init__(*args, **kwargs)
-        from htmd.home import home
-        from htmd.molecule.molecule import Molecule
-        from htmd.parameterization.util import canonicalizeAtomNames, getEquivalentsAndDihedrals
-        from yaml import load as yamlload
-        from pickle import load as pickleload
+    def setUp(self):
         self.refDir = home(dataDir='test-fftype')
-        self.Molecule = Molecule
```

Comment:
I may seem stupid to do this, but leaving it outside makes the tests slow, as for each individual test, it reloads the Molecule class (which is super slow). Maybe there's a cleaner way of doing this, but I found that adding it to the __init__ of the TestCase class worked

Reply:
As I showed, `Molecule` is imported only once.